docs/integrations/platforms/kubernetes.mdx

```diff
@@ -674,6 +678,51 @@ The namespace of the managed Kubernetes secret to be created.
 <Accordion title="managedSecretReference.secretType">
 Override the default Opaque type for managed secrets with this field. Useful for creating kubernetes.io/dockerconfigjson secrets.
 </Accordion>
+<Accordion title="managedSecretReference.template">
+Templates enable you to transform data from Infisical before storing it as a Kubernetes Secret.
+</Accordion>
+<Accordion title="managedSecretReference.template.includeAllSecrets">
+When set to true, this option injects all secrets retrieved from Infisical into your configuration. 
+Secrets defined in the template will override the automatically injected secrets.
+</Accordion>
+<Accordion title="managedSecretReference.template.data">
+Define secret keys and their corresponding templates. 
+Each data value uses a Golang template with access to all secrets retrieved from the specified scope.
+
+Secrets are structured as follows:
+```golang
+type TemplateSecret struct {
+    Value      string `json:"value"`
+    SecretPath string `json:"secretPath"`
+}
+```
+
+#### Example template configuration:
+```golang
```

Comment:
Why are we formatting as golang here, shouldn't it be yaml?

Reply:
Yes I agree it's YAML. I have changed to golang formatting because the context is used to explain the template option inside the YAML. Golang formatting highlights that part much better.